### PR TITLE
[133] Add GOV.UK Components Gem

### PR DESCRIPTION
**Pull request by aliuk2012**
### Context
PR #40 needs to be merged first.

### Changes proposed in this pull request
- Adds govuk-components gem
- Replaces our hard coded header with GOV.UK Header Component
- Replaces our hard coded skip links with GOV.UK Skip Link Component
- Replaces our hard coded link_to with govuk_link_to helper
- Completely removes our summary table in favour of GOV.UK Summary List Component

### CODE REVIEW

**Comment by aliuk2012** [comment]
[Add GOV.UK Components gem](https://trello.com/c/XK1Hg4RV/133-add-govuk-components-gem)

**Comment by SamJamCul** [comment]
There are two commits that need the ticket number prefixed!

**Comment by aliuk2012** [comment]
> There are two commits that need the ticket number prefixed!

@SamJamCul those are part of #40. The reason for not adding the ticket number is that they so

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -23,42 +23,15 @@
...
-            </ul>
-          </nav>
-        </div>
-      </div>
-    </header>
+    <%= render_component GovukComponent::Header.new( logo: 'GOV.UK',
+                                                     service_name: 'Register trainee teachers',
+                                                     service_name_href: "/") do |component| %>
+      <%= component.slot(:item, title: 'Navigation item 1', href: '/#page-1', active: true) %>
+      <%=component.slot(:item, title: 'Navigation item 2', href: '/#page-2') %>
```
Comment (gpeng) [inline app/views/layouts/application.html.erb:32]:
```suggestion
      <%= component.slot(:item, title: 'Navigation item 2', href: '/#page-2') %>
```

**Review by gpeng — APPROVED** [review]

**Review by darokel — APPROVED** [review]

**Pull request merged**